Helper/Api/Products.php

```diff
@@ -753,12 +756,24 @@ private function setGalleryImage($image, Product $target, $mainImage)
                 return true;
             } catch (LocalizedException $e) {
                 $this->logger->error($exception->getMessage(), $this->logger->buildReportData($exception));
+                return false;
             }
         } else {
             return false;
         }
     }
 
+    /**
+     * @param string $imageUrl
+     * @return string
+     */
+    private function getImageName(string $imageUrl): string
+    {
+        $path = parse_url($imageUrl, PHP_URL_PATH);
+
+        return basename($path);
+    }
```

Comment:
```suggestion
    private function getImageName(?string $imageUrl): string
    {
        if( !empty ( $imageUrl )) {
        
        $path = parse_url($imageUrl, PHP_URL_PATH);

        return basename($path);
        }
        return 'no_selection';
    }
```

looking at the comparisons in the rest of the code.

Reply:
did you also check the case that storekeeper product has no image?